odc/stac/eo3/_eo3converter.py

```diff
@@ -238,7 +238,7 @@ def _item_to_ds(
     if cfg is None:
         cfg = {}
 
-    md: RasterCollectionMetadata = getattr(product, "_md")
+    md: RasterCollectionMetadata | None = getattr(product, "_md", None)
```

Comment:
`: A|B` is py310+  syntax, we still support py39. Need to add `from __future__ import anotations` to the top of the file.

Reply:
Also I believe new datacube `Product` class doesn't allow setting extra attributes like `_md` in this case, so we need a different mechanism for passing on parsed collection metadata.

Reply:
The tests run just fine with both datacube 1.8.19 and 1.9, so I think it should be fine?